commands/general/stats.js

```diff
@@ -4,118 +4,165 @@ const moment = require('moment')
 
 module.exports.load = (client) => {
   client.commands['stats'] = {
-    run (message) {
-      let args = message.content.split(' ').splice(1)
-      let username
-      let discriminator
-      let mem
-      let userId
-      let av = './assets/default_avatar.jpg'
+    async run (message) {
+      let userInfo
+      try {
+        userInfo = selectTargetUser(message)
+      } catch (err) {
+        console.log(err.stack)
+        return client.errorMessage(message, `Unable to identify user: ${err}`)
+      }
+
+      try {
+        const [user, pos, tot, guild] = await Promise.all([
+          client.userRepository.load(userInfo.userId),
+          client.userRepository.getSessionPos(userInfo.userId),
+          client.userRepository.getSessionCount(),
+          client.guildRepository.load(message.guild.id)
+        ])
+        if (user !== null) {
+          const activeTime = getActiveTime(guild, userInfo.mem)
```

Comment:
Sorry I forgot to close the loop on the liveness discussion on the previous review. Writing to the db every minute sounds like a good idea anyway just to save ourselves from losing time if the host disappears on us again (though hopefully this will be less of a problem moving forward). Might it be a good idea to write anyway when `p!stats` or `p!lb` is called?

Doesn't have to happen here, as long as we're comfortable with the fact that we're somewhat regressing positional accuracy with this change (i.e. `p!stats` may claim you're in a different position from what `p!lb` says).